src/document.rs

```diff
@@ -507,6 +513,43 @@ impl Document {
             inner: self.inner.entry(k),
         }
     }
+
+    /// Attempts to encode the `Document` into a byte stream.
+    pub fn encode_document<W: Write + ?Sized>(&self, writer: &mut W) -> EncoderResult<()> {
```

Comment:
given that these are now methods on `Document`, we can drop it from the function name. So we can just call `document.encode()` or `Document::decode(reader)`.